datasource/src/main/scala/quasar/physical/s3/impl/isResource.scala

```diff
@@ -50,14 +51,15 @@ object isResource {
       false.pure[F]
     } else {
       // Don't use the metadata, just check the request status
-      client.status(request) >>= {
-        case Status.Ok => true.pure[F]
-        case Status.PartialContent => true.pure[F]
-        case Status.NotFound => false.pure[F]
-        case Status.RangeNotSatisfiable => false.pure[F]
-        case Status.Forbidden => Sync[F].raiseError(new Exception(s"Permission denied. Make sure you have access to the configured bucket"))
-        case s => Sync[F].raiseError(new Exception(s"Unexpected status returned during `isResource` call: $s"))
-      }
+      sign(request) >>= (r =>
+        client.status(r) >>= {
+          case Status.Ok => true.pure[F]
+          case Status.PartialContent => true.pure[F]
+          case Status.NotFound => false.pure[F]
+          case Status.RangeNotSatisfiable => false.pure[F]
```

Comment:
Why are these two cases returning `false` while the two below are raising errors? I'm wondering what criteria was used / should be used to distinguish between these cases.